nucleus/src/main/java/za/co/absa/subatomic/application/member/TeamMemberService.java

```diff
@@ -57,9 +57,28 @@ public String newTeamMember(String firstName, String lastName,
     }
 
     public String newTeamMemberFromSlack(String firstName, String lastName,
-            String email, String domainUsername, String screeName,
+            String email, String domainUsername, String screenName,
             String userId) {
+        TeamMemberEntity existingMember = this.findByEmail(email);
+        if (existingMember != null) {
+            throw new DuplicateRequestException(MessageFormat.format(
+                    "Requested email address {0} is already in use.",
+                    email));
+        }
 
+        existingMember = this.findByEmail(email);
```

Comment:
Should this not be findByDomainUsername?

Reply:
It should be

Reply:
Fixed

Reply:
Yes. Fixed now.